source/CScriptEngine.cpp

```diff
@@ -684,79 +684,67 @@ namespace CLEO
-            enum class VPref{ None, Game, User, Script, Cleo, Modules } virtualPrefix = VPref::None;
-            auto root = fsPath.begin();
-            if(root != fsPath.end())
-            {
-                if(*root == DIR_GAME) virtualPrefix = VPref::Game;
-                else if (*root == DIR_USER) virtualPrefix = VPref::User;
-                else if (*root == DIR_SCRIPT) virtualPrefix = VPref::Script;
-                else if (*root == DIR_CLEO) virtualPrefix = VPref::Cleo;
-                else if (*root == DIR_MODULES) virtualPrefix = VPref::Modules;
-            }
+            if(_strcmpi(r, DIR_GAME) == 0) virtualPrefix = VPref::Game;
+            else if (_strcmpi(r, DIR_USER) == 0) virtualPrefix = VPref::User;
+            else if (_strcmpi(r, DIR_SCRIPT) == 0) virtualPrefix = VPref::Script;
+            else if (_strcmpi(r, DIR_CLEO) == 0) virtualPrefix = VPref::Cleo;
+            else if (_strcmpi(r, DIR_MODULES) == 0) virtualPrefix = VPref::Modules;
+        }
 
-            // not virtual
-            if(virtualPrefix == VPref::None)
+        // not virtual
+        if(virtualPrefix == VPref::None)
+        {
+            if(fsPath.is_relative())
             {
-                if(fsPath.is_relative())
-                {
-                    if(customWorkDir != nullptr)
-                        fsPath = ResolvePath(customWorkDir) / fsPath;
-                    else
-                        fsPath = GetWorkDir() / fsPath;
-
-                    auto resolved = FS::weakly_canonical(fsPath).string();
-
-                    // ModLoader support: do not expand game dir relative paths
-                    if (resolved.find(Filepath_Root) == 0)
-                        return FS::relative(resolved, Filepath_Root).string();
-                    else
-                        return resolved;
-                }
-
-                return FS::weakly_canonical(fsPath).string();
+                if(customWorkDir != nullptr)
+                    fsPath = ResolvePath(customWorkDir) / fsPath;
+                else
+                    fsPath = GetWorkDir() / fsPath;
             }
 
-            // expand virtual paths
-            FS::path resolved;
+            auto result = fsPath.string();
+            NormalizeFilepath(result, false);
 
-            if (virtualPrefix == VPref::User) // user files location
+            // ModLoader support: keep game dir relative paths relative
+            if (result.length() > Filepath_Game.length() && // and separator
+                result[Filepath_Game.length()] == '\\' && // path separator after game path
+                StringStartsWith(GetWorkDir(), Filepath_Game, false) && // curent work dir is game root
+                StringStartsWith(result, Filepath_Game, false)) // resulting path is within game root
             {
-                resolved = GetUserDirectory();
+                result.replace(0, Filepath_Game.length() + 1, ""); // remove game root path prefix

```

Comment:
you can create a new util function: StripGamePath and result here and in loadPlugins

Reply:
Those are temporary fixes. Final goal is to always use absolute paths. User should not even need to do anything with path after receiving it from OPCODE_READ_PARAM_FILEPATH.